Add `else if` to language

cstlee opened the pull request:
The language now provides `else if` as syntactic sugar for an `if`
statement nested within an `else` block.

Close #14

cstlee committed:
Add `else if` to language

The language now provides `else if` as syntactic sugar for an `if`
statement nested within an `else` block.

Close #14

ongardie-sfdc commented on lib/parser.js:
```diff
@@ -613,7 +613,15 @@ let whileLoop = seqMap(keywords.while,
-  (keywords.else).then(block).times(0, 1).map((elses) => {
+  (keywords.else).then(alt(block, lazy(() => {
+    return ifElse;
+  }).source().map((elseIf) =>({
```
I think it'd be clearer if the entirety of ifElse was nested in a call to lazy(). Then you can omit the lazy() here.

cstlee replied:
Sure, I can pull the `.source().map()` into the `lazy` block.  I'm not sure I can omit the `lazy` though since `ifElse` is not yet defined at this point (we are in the middle of defining it).

ongardie-sfdc replied:
I mean to hoist the lazy call all the way up to `let ifElse = lazy(seqMap(...`

cstlee replied:
Oh, THAT kind of "entirety of ifElse."  Yep, will do.

ongardie-sfdc commented:
Thanks @cstlee. Can you also update `doc/LANGUAGE-GUIDE.md`?

cstlee committed:
Make ifElse "lazy" to allow clear recursion

Previously, only the recursive call to ifElse was wrapped in a lazy
function.  Now the entire ifElse function is lazy which hopefully
makes the recursive call more clear.

cstlee committed:
Update LANGUAGE-GUIDE to include `else if`

cstlee commented:
Addressed both the code and documentation change requests.  Let me know if there are other issues we should address.

ongardie-sfdc commented:
Thanks, merged.
